sway-book/src/documentation/language/program-types/predicate.md

```diff
@@ -0,0 +1,17 @@
+# Predicates
```

Comment:
In case it's useful, my attempt at explaining predicates on Fuel can be consulted [here](https://github.com/FuelLabs/sway-applications/tree/master/otc-swap-predicate#predicates-in-fuel)